src/connectdlg.cpp

```diff
@@ -220,15 +211,16 @@ void CConnectDlg::RequestServerList()
 
     // update list combo box (disable events to avoid a signal)
     cbxCentServAddrType->blockSignals ( true );
-    cbxCentServAddrType->setCurrentIndex ( static_cast<int> ( pSettings->eCentralServerAddressType ) );
+    cbxCentServAddrType->setCurrentIndex ( static_cast<int> ( pSettings->eCentralServerAddressType ) + pSettings->iCustomDirectoryIndex );
     cbxCentServAddrType->blockSignals ( false );
 
     // Get the IP address of the directory server (using the ParseNetworAddress
     // function) when the connect dialog is opened, this seems to be the correct
     // time to do it. Note that in case of custom directory server address we
     // use the first entry in the vector per definition.
```

Comment:
Again, the code no longer seems to agree with this comment, so something needs changing.